internal/manifests/collector/horizontalpodautoscaler_test.go

```diff
@@ -74,19 +80,31 @@ func TestHPA(t *testing.T) {
 			t.Run(test.name, func(t *testing.T) {
 				configuration := config.New()
 				params := manifests.Params{
-					Config:  configuration,
-					OtelCol: otelcol,
-					Log:     logger,
+					Config: configuration,
+					OtelCol: v1alpha1.OpenTelemetryCollector{
+						ObjectMeta: otelcol.ObjectMeta,
+						Spec: v1alpha1.OpenTelemetryCollectorSpec{
+							MinReplicas: otelcol.Spec.OpenTelemetryCommonFields.Autoscaler.MinReplicas,
+							MaxReplicas: otelcol.Spec.OpenTelemetryCommonFields.Autoscaler.MaxReplicas,
+							Autoscaler: &v1alpha1.AutoscalerSpec{
+								TargetCPUUtilization:    otelcol.Spec.OpenTelemetryCommonFields.Autoscaler.TargetCPUUtilization,
+								TargetMemoryUtilization: otelcol.Spec.OpenTelemetryCommonFields.Autoscaler.TargetMemoryUtilization,
+							},
+						},
+					},
+					Log: logger,
 				}
-				raw := HorizontalPodAutoscaler(params)
+				raw, err := HorizontalPodAutoscaler(params)
+				require.NoError(t, err)
 
 				hpa := raw.(*autoscalingv2.HorizontalPodAutoscaler)
 
 				// verify
 				assert.Equal(t, "my-instance-collector", hpa.Name)
 				assert.Equal(t, "my-instance-collector", hpa.Labels["app.kubernetes.io/name"])
-				assert.Equal(t, int32(3), *hpa.Spec.MinReplicas)
-				assert.Equal(t, int32(5), hpa.Spec.MaxReplicas)
+				// require.NotNil(t, hpa.Spec.MinReplicas)
```

Comment:
Cam this be removed?

Reply:
Why should it be removed? The goal is to of this PR is to only prepare the switch from v1alpha1 to v1alpha2 in `internal/manifests/collector`. Since this is quite a big change, I want to avoid changing all kinds of logic, I do not have to touch.

Reply:
Because the line is commented. So, it seems it is not needed anymore.